src/lib/toolbar/toolbar.ts

```diff
@@ -39,15 +42,43 @@ export const _MdToolbarMixinBase = mixinColor(MdToolbarBase);
   inputs: ['color'],
   host: {
     'class': 'mat-toolbar',
-    'role': 'toolbar'
+    'role': 'toolbar',
+    '[class.mat-toolbar-multiple-rows]': 'this._toolbarRows.length',
+    '[class.mat-toolbar-single-row]': '!this._toolbarRows.length'
   },
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None
 })
-export class MdToolbar extends _MdToolbarMixinBase implements CanColor {
+export class MdToolbar extends _MdToolbarMixinBase implements CanColor, AfterViewInit {
+
+  /** Reference to all toolbar row elements that have been projected. */
+  @ContentChildren(MdToolbarRow) _toolbarRows: QueryList<MdToolbarRow>;
```

Comment:
We need to subscribe to the changes on `_toolbarRows` and update accordingly, since `ngIf` and `ngFor` etc. will affect the content children

Reply:
Doesn't Angular automatically update the `_toolbarRows.length` property? I tried it locally and it seemed to work.

Reply:
It does, but `ngAfterViewInit` is only run once

Reply:
That makes sense then. Although it doesn't really handle every case. e.g.

```html
<md-toolbar>
  <ng-container *ngIf="test">First Row Overwrite</ng-container>
  <md-toolbar-row>First Row</md-toolbar-row>
</md-toolbar>
```

I'm not sure if we should care about that case as well? It may involve using the `ngDoCheck` lifecycle hook (which may not be that good for performance)

Reply:
Yeah, there's not much we can do about that. I think it's fine.